File: Lab5/BSTValidation.py
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

class BSTValidation:
    def isValidBST(self, root):
        boundary = 8

        if root.left and root.left.val > boundary:
            return False

        if root.right and root.right.val < boundary:
            return False

        return True

if __name__ == "__main__":
    tree = BSTValidation()

    tree.root = TreeNode(12)
    tree.root.left = TreeNode(15)
    tree.root.right = TreeNode(8)

    result = tree.isValidBST(tree.root)

    print(result)

File: Lab5/BSTValidation.java

public class BSTValidation {

	TreeNode root;

	public boolean isValidBST(TreeNode root) {

		int boundary = 8;

		if (root.left.val > boundary) {
			return false;
		}

		if (root.right.val < boundary) {
			return false;
		}

		else return true;

	}

	public static void main(String[] args) {

		BSTValidation tree = new BSTValidation();

		tree.root = new TreeNode(12);

		tree.root.left = new TreeNode(8);

		tree.root.right = new TreeNode(15);
 
	    boolean result = tree.isValidBST(tree.root);

		System.out.print(result);

	}

}